docs/NooBaaNonContainerized/Configuration.md

```diff
@@ -0,0 +1,321 @@
+* <u>Example</u>:
+    ```sh
+    > ls -la /etc/noobaa.conf.d/access_keys/
+    ABCABCABCABCABCABCAB.symlink -> ../accounts/alice.json
+    DEFDEFDEFDEFDEFDEFDE.symlink -> ../accounts/bob.json
+    GHIGHIGHIGHIGHIGHIGH.symlink -> ../accounts/charlie.json
+    ```
+`buckets/`
+* <u>Type</u>: Directory.
+* <u>Required</u>: True.
+* <u>Description</u>: A directory that contains configuration files for individual buckets, each bucket configuration file is named {bucket_name}.json and adheres to the [bucket schema](../../src/server/system_services/schemas/nsfs_bucket_schema.js).
+* <u>Example</u>:
+    ```sh
+    > ls /etc/noobaa.conf.d/buckets/
+    bucket1.json
+    bucket2.json
+    bucket3.json
+    ```
+`config.json`
+* <u>Type</u>: File.
+* <u>Required</u>: False.
+* <u>Description</u>: A JSON file that contains shared configurations of all nodes on the cluster and machine specific configurations, for more details check - [Config File Customizations](#config-file-customizations)
+* <u>Example</u>: 
+    ```JSON
+    {
+        "ENDPOINT_FORKS": 2,
+        "host_customization": {
+            "{node_name}" : {
+                "ENDPOINT_FORKS": 3, 
+            }
+        }
+    }
+    ```
+
+`master_keys.json`
+* <u>Type</u>: File.
+* <u>Required</u>: False.
+* <u>Description</u>: A JSON file containing master keys information. The master keys specified in this file used for encryption of sensitive data, such as account's secret key. When using 'executable' master keys store type, the master_keys.json is stored on a different location, therefore this file is optional.
+* <u>Example</u>: 
+    ```JSON
+    {
+        "timestamp":1719143407465,
+        "active_master_key":"aaaaaaaa",
+        "master_keys_by_id": {
+            "aaaaaaaa": {
+                "id":"aaaaaaaa",
+                "cipher_key":"bbbbbbbb",
+                "cipher_iv":"cccccccc",
+                "encryption_type":"aes-256-gcm"
+            }
+        }
+    }
+    ```
+
+`certificates/`
+* <u>Type</u>: Directory.
+* <u>Required</u>: False.
+* <u>Description</u>: A directory that stores SSL/TLS certificates for secure communication between S3 client and NooBaa Service. For more info, See - [Certificates](#certificates).
+* <u>Example</u>: 
+    ```sh
+    > ls /etc/noobaa.conf.d/certificates/
+    tls.crt
+    tls.key
+    ```
+
+`config_dir_redirect`
+* <u>Type</u>: File.
+* <u>Required</u>: False.
+* <u>Description</u>: A fixed location for setting a custom configuration directory location. For more info, See - [Custom Configuration Directory Setup](#custom-configuration-directory-setup).
+* <u>Example</u>: 
+    ```sh
+    > cat /etc/noobaa.conf.d/config_dir_redirect
+    /path/to/custom/config/dir/
+    ```
+
+## Config File Customizations
+
+One can customize NooBaa by creation of config.json file under the configuration directory - `/path/to/config_dir/config.json`.    
```

Comment:
Minor suggestion
```suggestion
You can customize NooBaa by creation of config.json file under the configuration directory - `/path/to/config_dir/config.json`.    
```

Reply:
I prefer not using `you` in formal documentations, I changed it to `a user` although it's generally fine using `One` as well.